src/main/java/ch/bfh/bti7081/s2018/green/presenters/MedicationPrescriptionPresenter.java

```diff
@@ -0,0 +1,123 @@
+
+        btnEdit.addClickListener(click -> {
+            btnEdit.setCaption("Cancel");
+            this.addAllComponents(true);
+            //this.addCancleButton();
+            this.addCancelAction(btnEdit);
+            this.addSaveButton();
+        });
+
+        this.addComponent(btnEdit, "medEditButton");
+    }
+    private void addCreateNewButtonListener (String btnText) {
+
+        this.addSaveNewEntryButton();
+        this.addCancelButton();
+    }
+
+    /*
+     * Adds cancel button. OnClick protectes the fields and reloads the data from object
+     */
+    private void addCancelAction(Button btnCancelEdit) {
+        btnCancelEdit.addClickListener(click ->{
+            this.addTheComponents(false);
+            addEditButtonListener("Edit");
+            this.removeComponent("medSaveButton");
+
+        });
+    }
+
+    /*
+     * Adds a save button, the function is not implementend yet
+     */
+    private void addSaveButton () {
+        Button btnSave = new Button("Save");
+        btnSave.addClickListener(click -> {
+            this.saveDataChanges();
+            // maybe display "your data has been saved" notification here?
+            //Notification.show("Your changes have been saved (not really, just a test)");
+        });
+        this.addComponent(btnSave, "medSaveButton");
+    }
+
+    private void addSaveNewEntryButton () {
+        Button btnSave = new Button("Save");
+        btnSave.addClickListener(click -> {
+            this.saveNewData();
+            // maybe display "your data has been saved" notification here?
+            //Notification.show("New record has been saved (not really, just a test)");
+        });
+        this.addComponent(btnSave, "medEditButton");
+    }
+
+    private void addCancelButton () {
+        Button btnCancel = new Button("Cancel");
+        btnCancel.addClickListener(click-> {
+            this.window.close();
+        });
+        this.addComponent(btnCancel, "medCancelEditButton");
+    }
+
+    private void saveNewData (){
+
+        System.out.println("Medi Name:" + getMedName().getActualValue());
+        System.out.println("Start Date: " + getMedStartDate().getActualValue());
+        System.out.println("End Date: " + getMedEndDate());
+        System.out.println("Period: " + getMedPeriod().getActualValue());
+        System.out.println("Dose: " + getMedDose().getActualValue());
+
+
+        Medication newMed = new Medication((String) getMedName().getActualValue(),
+                (LocalDateTime) getMedStartDate().getActualValue(),
+                (LocalDateTime) getMedEndDate().getActualValue(),
+                Integer.parseInt((String) getMedPeriod().getActualValue()),
+                Float.parseFloat((String) getMedDose().getActualValue()),
+                this.data.getCurrentStaff(),
+                this.data.getCurrentPatient());
+
+        EntityManagerFactory emf = Persistence.createEntityManagerFactory("pmsDB");
```

Comment:
use manager

Reply:
Need explanation maybe. So put in sprint 3 since it isn't a too complex task.